src/pkg/podman/podman.go

```diff
@@ -227,9 +227,16 @@ func IsToolboxImage(image string) (bool, error) {
 }
 
 // Pull pulls an image
-func Pull(imageName string) error {
+//
+// authfile is a path to a JSON authentication file and is internally used only
+// if it is not an empty string. The file at the path should exist.
```

Comment:
Strictly speaking, nobody can ensure that *the file at the path should exist*.  All filesystem operations are inherently racy - files can come and go behind under our feet at any moment.  That's why the only option is to go ahead and try the operation and be ready to handle the errors.

`podman pull` does the same, but it doesn't offer an unique exit code for us to detect the specific error and show a meaningful message to the user.  We should try to see if we can get it to do that.

In the meantime, we can get away with checking for the existence of the file beforehand, because it will catch the vast majority of cases.  However, in terms of documenting this API, we can't expect the caller to guarantee that the file should exist.